### PR TITLE
fix: return raw responses when requested even with 0 records

### PR DESCRIPTION
**Pull request by aaronsteers**
# fix: return raw responses when requested even with 0 records

## Summary
Fixes a bug in `execute_stream_test_read` where `include_raw_responses_data: true` returned `null` instead of raw API responses when no records were found (e.g., `max_records=0`).

The issue occurred because:
1. When `max_records=0`, the function would fail with "No API output returned" and return early without processing raw responses
2. The successful path had a condition `if include_raw_responses_data is True and slices:` that prevented raw responses from being returned when `slices` was empty

**Changes made:**
- Added raw response handling to the early return path when `success=False`
- Removed the `and slices` condition so raw responses are returned whenever explicitly requested
- Raw responses now return empty list `[]` instead of `null` when no data is available but responses were requested

## Review & Testing Checklist for Human
- [ ] **Verify the original bug is fixed**: Test with `max_records=0` and `include_raw_responses_data=true` to confirm raw responses are returned (not null)
- [ ] **Test edge cases**: Try different failure scenarios (invalid manifests, network errors, etc.) with raw responses enabled to ensure the early return path works correctly
- [ ] **Check API contract compatibility**: Verify that existing callers can handle the new `raw_api_responses` field in failed responses without breaking
- [ ] **Test successful scenarios**: Confirm that removing the `and slices` condition doesn't cause issues when slices is legitimately empty in successful operations

### Notes
- The fix addresses both the early return path (when stream test fails) and the successful path (when stream test succeeds but has empty slices)
- All existing tests pass, but manual testing of the specific reported scenario is recommended
- Session requested by AJ Steers (@aaronsteers)
- Link to Devin run: https://app.devin.ai/sessions/98be4b20044e45f390a6036aa29f995b

### CODE REVIEW

File: connector_builder_mcp/validation_testing.py [hunk 1]
```diff
@@ -401,11 +401,19 @@ def execute_stream_test_read(  # noqa: PLR0914
     stream_data: dict[str, Any] = {}
     if result.record and result.record.data:
         stream_data = result.record.data
+        slices_from_stream = stream_data.get("slices", [])
+        if (
+            include_raw_responses_data
+            and not slices_from_stream
+            and "auxiliary_requests" in stream_data
```
Comment (aaronsteers) [inline connector_builder_mcp/validation_testing.py:408]: Devin, please add a comment to explain auxiliary requests

**Review by aaronsteers — COMMENTED** [review]
Devin, One comment inline. Also, your code is extremely ugly and hard to read. And I have a feeling you are still "doing too much". Move parsing into helper function (s) if you need to but please make this less ugly. 

You added hundreds of lines of code for what I expected in 2-10 lines
